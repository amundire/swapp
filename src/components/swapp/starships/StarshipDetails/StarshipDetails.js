import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { STARSHIP_QUERY } from '../../../../api/swapp/starship-requester';
import { useParams } from 'react-router-dom';
import Starship from '../../starships/Starship/Starship';

const CharacterDetails = () => {
    const { id } = useParams();

    const { data, loading, error } = useQuery(STARSHIP_QUERY, {
        variables: { starshipId: id },
    });

    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error getting starship data</p>;

    const { starship } = data;

    //todo: add starships

    return (
        <section id="characterDetails">
            <div className="characterDetailsLeft">
                {JSON.stringify(starship)}
            </div>
            <div className="characterDetailsRight">
                RADAR CHART
            </div>
        </section>
    )
}

export default CharacterDetails;