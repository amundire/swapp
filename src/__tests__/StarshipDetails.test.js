import React from 'react';
import { shallow } from 'enzyme';
import StarshipDetails from '../components/swapp/starships/StarshipDetails/StarshipDetails';

describe('<StarshipDetails />', () => {
    it('renders without crashing', () => {
        const data = {
            starship: {
                id: 'starships.1',
                name: 'X-wing',
                model: 'X-wing',
                starshipClass: 'starfighter',
                image: 'https://static.turbosquid.com/Preview/2015/10/14__02_29_23/xwingtopleft_01_open_01.jpgb5dc9c7c-25bc-44f8-88ba-50e41873111aOriginal.jpg',
                cost: 149999,
                maxAtmosphericSpeed: 1050,
                maxMLPerHour: 100,
                hyperdriveRating: 1,
                crew: 1,
            },
        },
            wrapper = shallow(<StarshipDetails data={data} />);
        expect(wrapper).toMatchSnapshot();
    });
});
