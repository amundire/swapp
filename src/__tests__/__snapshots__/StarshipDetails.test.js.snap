// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarshipDetails /> renders without crashing 1`] = `
<section
  id="starshipDetails"
>
  <p
    className="starshipTitle"
  >
    X-wing
  </p>
  <p
    className="starshipModel"
  >
    X-wing
  </p>
  <div
    className="StarshipDivider"
  >
    <div
      className="starshipDetailsLeft"
    >
      <p
        className="starshipName"
      >
        X-wing
      </p>
      <img
        alt="starship image"
        src="https://static.turbosquid.com/Preview/2015/10/14__02_29_23/xwingtopleft_01_open_01.jpgb5dc9c7c-25bc-44f8-88ba-50e41873111aOriginal.jpg"
      />
      <p
        className="starshipStat"
      >
        Class: 
        <span
          className="starshipStatParam"
        >
          starfighter
        </span>
      </p>
      <p
        className="starshipStat"
      >
        Cost: 
        <span
          className="starshipStatParam"
        >
          149999
        </span>
      </p>
      <p
        className="starshipStat"
      >
        Crew: 
        <span
          className="starshipStatParam"
        >
          1
        </span>
      </p>
      <p
        className="starshipStat"
      >
        Max atmospheric speed: 
        <span
          className="starshipStatParam"
        >
          1050
        </span>
      </p>
      <p
        className="starshipStat"
      >
        Hyperdrive rating: 
        <span
          className="starshipStatParam"
        >
          1
        </span>
      </p>
    </div>
    <div
      className="starshipDetailsRight"
    >
      <RadarChartComponent
        starship={
          Object {
            "cost": 149999,
            "crew": 1,
            "hyperdriveRating": 1,
            "id": "starships.1",
            "image": "https://static.turbosquid.com/Preview/2015/10/14__02_29_23/xwingtopleft_01_open_01.jpgb5dc9c7c-25bc-44f8-88ba-50e41873111aOriginal.jpg",
            "maxAtmosphericSpeed": 1050,
            "maxMLPerHour": 100,
            "model": "X-wing",
            "name": "X-wing",
            "starshipClass": "starfighter",
          }
        }
      />
    </div>
  </div>
</section>
`;
